app/controllers/categories_controller.rb

```diff
@@ -1,3 +1,81 @@
 class CategoriesController < ApplicationController
   
+  def index
+    @categories = Category.all
+  end
+  
+  def new
+    @categories  = Category.new
+  end
+  
+  def create
+    @category = Category.new(category_params) 
+    if @category.save 
+      flash[:success] = "Category added successfully"
+      redirect_to  root_path #merchant_path(@merchant)
+      return
+    else 
+      flash.now[:failure] = "Category failed to save"
+      render :new, status: :bad_request 
+      return
+    end
+  end
+  
+  # def show
+  #   category_id = params[:id]
+  #   @category = Category.find_by(id: category_id)
+    
+  #   if @category.nil?
+  #     head :not_found
+  #     return
+  #   end     
+  # end
+  
+  def edit
+    @category = Category.find_by(id: params[:id])
+    
+    if @category.nil?
+      head :not_found
+      return
+    end
+  end
+  
+  def update
+    @category = Category.find_by(id: params[:id])
+    
+    if @category.nil?
+      head :not_found
+      return
+    end
+
+    if @category.update(category_params)
+      redirect_to root_path 
+      return
+    else 
+      render :edit, status: :bad_request 
+      return
+    end
+  end
+  
+  def destroy
+    @category.destroy
```

Comment:
You will have to first find the category that you want to destroy. If you merge my suggestion from above with the before_action and add a `find_category` private method, you will be able to find the category that the user selects to be destroyed.